### PR TITLE
Updated to render tables for Vue only

### PR DESCRIPTION
**Pull request by kirlat**
This update eliminates old code that renders inflection tables using HTML nodes. There were also several changes to optimize rendering performance, mostly in reducing number of operations required to be made before an inflection table is displayed. 

It worked well in my tests, but probably a more broad testing with more varied scenarios is required (but I hope it will work well there too!).

Please let me know what do you think.

### CODE REVIEW

File: tools/inflections-xml2csv/converter.js [hunk 1]
```diff
@@ -1644,8 +1644,9 @@ try {
         let j = xmlToJSON.parseString(d)
         json['infl-data'] = json['infl-data'].concat(j['infl-data'])
       }
-      writeData(posCfg.suffixes.get(json), posCfg.suffixes.outputPath)
-      writeData(posCfg.footnotes.get(json), posCfg.footnotes.outputPath)
+      // Suffixes were edited manually. Running this conversion would erase those changes
```
Comment (balmas) [inline tools/inflections-xml2csv/converter.js]: Ah, I didn't realize we were still running the conversions. Maybe we should stop that for all tables, and make the csv files the official data files now?

File: views/lib/view.js [hunk 2]
```diff
@@ -31,10 +32,12 @@ export default class View {
     this.id = 'base_view'
     this.name = 'base view'
     this.title = 'Base View'
-    this.hasComponentData = false // True if vue supports Vue.js components
+    this.hasPrerenderedTables = false // True if vue supports Vue.js components
```
Comment (balmas) [inline views/lib/view.js]: very minor: the comment needs to be updated or removed

File: views/lib/narrow-view.js [hunk 3]
```diff
@@ -7,28 +7,20 @@ import NarrowViewGroup from './narrow-view-group'
...
-    this.headers = headers
-    this.titleColumnQty = titleColumnQty
+  constructor (table) {
+    this.table = table
     this.groups = []
-    this.groupQty = groupQty
     this.groupSize = 0
-    if (groupQty) {
-      this.groupSize = this.columns.length / groupQty
+    if (this.table.features.firstColumnFeature.size) {
```
Comment (balmas) [inline views/lib/narrow-view.js]: can you add a comment that explains what this is doing?

File: views/lib/view.js [hunk 4]
```diff
@@ -46,6 +49,28 @@ export default class View {
      * @type {string}
      */
     this.creditsText = ''
+
+    this.initialized = false
+  }
+
+  /**
+   * Converts an InflectionData, returned from an inflection tables library, into an HTML representation of an inflection table.
```
Comment (balmas) [inline views/lib/view.js]: again, minor but can we update the comment to reflect what is actually happening now? It's not producing HTML anymore but a data object of rows and columns, correct?

File: views/lib/view.js [hunk 5]
```diff
@@ -162,14 +187,17 @@ export default class View {
   /**
    * Converts an InflectionData, returned from an inflection tables library, into an HTML representation of an inflection table.
    * `messages` provides a translation for view's texts.
+   * @param {Object} options - Render options
```
Comment (balmas) [inline views/lib/view.js]: See above, can you update the description of the render method to reflect what it is currently doing?

File: tools/inflections-xml2csv/converter.js (hunk 1 above)
Comment (kirlat) [reply tools/inflections-xml2csv/converter.js]: We're usually running this script once for each new piece of data. But I will probably comment out all conversion routines for safety.

File: views/lib/view.js (hunk 2 above)
Comment (kirlat) [reply views/lib/view.js]: updated

File: views/lib/narrow-view.js (hunk 3 above)
Comment (kirlat) [reply views/lib/narrow-view.js]: Good idea! Updated.

File: views/lib/view.js (hunk 5 above)
Comment (kirlat) [reply views/lib/view.js]: Sure, done!

**Comment by kirlat** [comment]
Pushed an updated version. Thanks for comments!

File: lib/lang/latin/data/verb/suffixes.csv [hunk 6]
```diff
@@ -706,4 +706,4 @@ tus esse,3rd,passive,infinitive,perfect,,,,regular,
 ātum īrī,1st,passive,infinitive,future,,,,regular,
 itum īrī,2nd,passive,infinitive,future,,,,regular,
 tum īri,3rd,passive,infinitive,future,,,,regular,
-ītum īrī,4th,passive,infinitive,future,,,,regular,
+ītum īrī,4th,passive,infinitive,future,,,,regular
```
Comment (balmas) [inline lib/lang/latin/data/verb/suffixes.csv]: actually we need the trailing comma to indicate that we have an empty column. See my comments in #107 which will fix this via #108

**Comment by balmas** [comment]
Please note I did introduce a bug with my PR for table data see #107 (and PR #108). I'm not sure it's the same issue but definitely problematic.

**Comment by kirlat** [comment]
@irina060981, thanks for catching this! This is a situation when there are no suffix matches at all, and, because we hide columns with no suffix matches by default, nothing is shown. If to press "Full table" button, data columns will become visible.

I think a best solution would be to show all columns in such situations by default and hide a "Collapse/Full table" button because such tables cannot be collapsed or expanded. @balmas, @irina06098, what do you think?

**Comment by irina060981** [comment]
@kirlat 
yes, I agree with your suggestion - I think it will be a clear behaviour pf the UI.

**Comment by balmas** [comment]
Yes, I agree. If no suffix matches at all, then show the full table.

**Comment by kirlat** [comment]
Updated so that tables with no suffix matches will always be shown in full width and with "Collpase/Full table" button hidden.
@balmas, @irina060981 By now all outstanding issues should be fixed. Please review and let me know if anything remained. Thanks!

**Comment by balmas** [comment]
@kirlat this looks fine to me. I think we still need to add the trailing comma back in the verb suffixes table, but I am working on that file anyway so I'll just put it back in. Thanks!

**Comment by kirlat** [comment]
@balmas, oops, forgot about trailing comma. Returned it back. Thanks for reminding!

**Comment by irina060981** [comment]
@kirlat, I think that it is really ok now - besides only one little problem (it was registered an issue before and it is not solved for now in this version)
https://github.com/alpheios-project/components/issues/102

It looks like this:

1) Open an inflections tab for some word that has inflections (for example **beatum**)
It shows correctly (and really fast)

2) If you try to open a word without inflection data (for example any persian word) - the inflection's tab won't become clear - it stays to show previous inflections for **beatum**

**Pull request closed without merging**